### PR TITLE
Increasing allowed wasm bytecode size

### PR DESCRIPTION
**Pull request by nicolaslara**
Closes: #4171 

## What is the purpose of the change

Increase the max allowed wasm size to 2MB

## Brief Changelog

  - Increased max allowed wasm size to 2MB

## Testing and Verifying

Added an E2E test uploading a large file. The test fails without the change.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Comment by Magicloud** [comment]
I tested with our contract. It works.

**Review by ValarDragon — APPROVED** [review]
LGTM, but needs state breaking changelog. (And you can update #4170 )

File: app/app.go [hunk 1]
```diff
@@ -158,6 +160,14 @@ func initReusablePackageInjections() {
 	}
 }
 
+// overrideWasmVariables overrides the wasm variables to:
+//   - allow for larger wasm files
+func overrideWasmVariables() {
+	// Override Wasm size limitation from WASMD.
+	wasmtypes.MaxWasmSize = 2 * 1024 * 1024
+	wasmtypes.MaxProposalWasmSize = wasmtypes.MaxWasmSize
+}
```
[the remark this answers is not in the document]
Comment (nicolaslara) [reply app/app.go:169]: yeah, I feel like wasmd configs are all over the place, but I don't know if there's a reason for that. For example, I often use a local fork to override `defaultContractDebugMode` (using --trace also adds that, but I think that doesn't work on tests and I don't want other side effects of adding trace).

But maybe there's something I'm missing. Do you know the right way to do this with `apply()`? Is that something we can define on osmosis or would it need to change in wasmd?

[the remark this answers is not in the document]
Comment (nicolaslara) [reply app/app.go:169]: added https://github.com/osmosis-labs/osmosis/issues/4206 to reorganize this in the future.

**Pull request merged**